api/v1alpha1/api.go

```diff
@@ -211,6 +211,18 @@ type AIGatewayRouteRule struct {
 	// Please refer to https://gateway.envoyproxy.io/docs/tasks/traffic/failover/ as well as
 	// https://gateway.envoyproxy.io/docs/tasks/traffic/retry/.
 	//
+	// When multiple backends are specified, this must not contain backends with kind=InferencePool. (TODO: enforce this in the validation.)
```

Comment:
Have no idea here whether the InferencePool refers to the `gateway-api-inference-extension` specifically or any consumer that will pick the endpoint themselves, if the latter, I think we should polish the name of the backend type here.